src/cmd-cloud-prune

```diff
@@ -0,0 +1,300 @@
+            current_builds_json = json.load(f)
+        merged_builds_json = current_builds_json | remote_builds_json_data
+        if not dry_run:
+            # Make sure we have the merged json as local builds/builds.json
+            save_builds_json(merged_builds_json)
+        # Upload the local builds.json to s3
+        s3_copy(s3_client, BUILDFILES['list'], bucket, f'{prefix}/builds.json', CACHE_MAX_AGE_METADATA, acl, extra_args={}, dry_run=dry_run)
+    else:
+        s3_copy(s3_client, BUILDFILES['list'], bucket, f'{prefix}/builds.json', CACHE_MAX_AGE_METADATA, acl, extra_args={}, dry_run=dry_run)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(prog="coreos-assembler cloud-prune")
+    parser.add_argument("--policy", required=True, type=str, help="Path to policy YAML file")
+    parser.add_argument("--dry-run", help="Don't actually delete anything", action='store_true')
+    parser.add_argument("--upload-builds-json", help="Push builds.json", action='store_true')
+    parser.add_argument("--stream", type=str, help="CoreOS stream", required=True)
+    parser.add_argument("--gcp-json-key", help="GCP Service Account JSON Auth", default=os.environ.get("GCP_JSON_AUTH"))
+    parser.add_argument("--gcp-project", help="GCP Project name", default=os.environ.get("GCP_PROJECT_NAME"))
+    parser.add_argument("--acl", help="ACL for objects", action='store', default='private')
+    parser.add_argument("--aws-config-file", default=os.environ.get("AWS_CONFIG_FILE"), help="Path to AWS config file")
+    return parser.parse_args()
+
+
+def prune_cloud_uploads(build, cloud_config, dry_run):
+    # Ensure AWS AMIs and GCP images are removed based on the configuration
+    errors = []
+    errors.extend(deregister_aws_amis(build, cloud_config, dry_run))
+    errors.extend(delete_gcp_image(build, cloud_config, dry_run))
+
+    if errors:
+        print(f"Found errors when removing cloud-uploads for {build.id}:")
+        for e in errors:
+            print(e)
+        raise Exception("Some errors were encountered")
+
+
+def deregister_aws_amis(build, cloud_config, dry_run):
+    errors = []
+    aws_credentials = cloud_config.get("aws", {}).get("credentials")
+    for ami in build.images.get("amis", []):
+        region_name = ami.get("name")
+        ami_id = ami.get("hvm")
+        snapshot_id = ami.get("snapshot")
+        if dry_run:
+            print(f"Would delete {ami_id} and {snapshot_id} for {build.id}")
+            continue
+        if ami_id and snapshot_id and region_name:
+            try:
+                deregister_aws_resource(ami_id, snapshot_id, region=region_name, credentials_file=aws_credentials)
+            except Exception as e:
+                errors.append(e)
+        else:
+            errors.append(f"Missing parameters to remove {ami_id} and {snapshot_id}")
+    return errors
+
+
+def delete_gcp_image(build, cloud_config, dry_run):
+    errors = []
+    gcp = build.images.get("gcp")
+    if gcp:
```

Comment:
Minor: can play the same trick here of handling the trivial case first and returning early, then dedent.